internal/gather/gather_bytes.go

```diff
@@ -114,26 +114,57 @@ func (b Bytes) ReadAt(p []byte, off int64) (n int, err error) {
 	return len(p), b.AppendSectionTo(bytes.NewBuffer(p[:0]), int(off), len(p))
 }
 
-// Reader returns a reader for the data.
-func (b Bytes) Reader() io.Reader {
-	b.assertValid()
+type bytesReadSeekCloser struct {
```

Comment:
`bytes.Reader` already implements `io.ReadSeeker`. It's only missing `Close()`. Why not extend `bytes.Reader` with just `Close()`? What am I missing?

Reply:
there's still a case of MultiReader, which don't implement io.ReadSeeker unfortunately.